shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java

```diff
@@ -43,24 +46,31 @@
  * Shell interpreter for Zeppelin.
  */
 public class ShellInterpreter extends Interpreter {
-  Logger logger = LoggerFactory.getLogger(ShellInterpreter.class);
-  private static final String EXECUTOR_KEY = "executor";
-  public static final String SHELL_COMMAND_TIMEOUT = "shell.command.timeout.millisecs";
-  int commandTimeOut;
-  private static final boolean isWindows = System
-          .getProperty("os.name")
-          .startsWith("Windows");
-  final String shell = isWindows ? "cmd /c" : "bash -c";
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShellInterpreter.class);
+  private static final String SHELL_COMMAND_TIMEOUT = "shell.command.timeout.millisecs";
+  private static final String DEFAULT_COMMAND_TIMEOUT = "600000";
+  private final boolean isWindows = System.getProperty("os.name").startsWith("Windows");
+  private final String shell = isWindows ? "cmd /c" : "bash -c";
+  private Map<String, DefaultExecutor> executors;
+
+  static {
+    Interpreter.register(
+      "sh", "sh",
+      ShellInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+        .add(SHELL_COMMAND_TIMEOUT, DEFAULT_COMMAND_TIMEOUT,
+          "Shell command time out in millisecs. Default = " + DEFAULT_COMMAND_TIMEOUT).build()
+    );
+  }
```

Comment:
Is there a reason to use static interpreter registration mechanism here, aftert [ZEPPELIN-923](https://issues.apache.org/jira/browse/ZEPPELIN-923) AKA #956 ?